### ISSUE TITLE
Kubernetes native sidecar not supported

### ISSUE THREAD

**Issue by YvesZelros** [Azure/azure-workload-identity#1312]
**Describe the bug**

AWI remove the field **_restartPolicy_** on init containers that break the support of [Kubernetes native sidecar](https://kubernetes.io/blog/2023/08/25/native-sidecar-containers/)

**Steps To Reproduce**

```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: simple-deployment
spec:
  replicas: 1
  selector:
    matchLabels:
      app: simple-deployment
  template:
    metadata:
      labels:
        app: simple-deployment
        azure.workload.identity/use: "true"
    spec:
      initContainers:
        - command:
            - "/bin/sh"
            - "-c"
            - "while true; do echo 'hello from init'; sleep 1; done"
          image: busybox
          name: init
          restartPolicy: Always
      containers:
        - command:
            - "/bin/sh"
            - "-c"
            - "while true; do echo 'hello from main'; sleep 1; done"
          image: busybox
          name: main
```
**Expected behavior**

The main container must be running and not waiting the end of the init container as the init container is a native sidecar (restartPolicy = Always).

**Logs**
NAME                                 READY   STATUS     RESTARTS   AGE
simple-deployment-7854b77d4b-hbs7r   0/1     Init:0/1   0          3m35s

**Environment**
- Kubernetes version (use `kubectl version`): Server Version: v1.29.2
- Cloud provider or hardware configuration: AKS

**Additional context**
https://github.com/istio/istio/issues/49221

**Comment by aramase** [comment]
Thanks for reporting the issue. I have a PR open to update the deps: https://github.com/Azure/azure-workload-identity/pull/1292.

**Comment by eplightning** [comment]
Any chance of getting a new release soon and deploying that on AKS? This is blocking us from using new sidecar functionality.

**Comment by mikec-greene** [comment]
Is it possible to get an ETA on when we might see this appearing in AKS? We're also blocked from using native sidecars in Istio, many thanks.

**Comment by rene-bos** [comment]
The PR which fixes this issue (#1292) has already been released in [v1.3.0](https://github.com/Azure/azure-workload-identity/releases/tag/v1.3.0) on the 6th of June. So you can already install it in your AKS cluster 😄.

**Comment by mikec-greene** [comment]
Great thanks @rene-bos! I missed it in the changelog. We use the integrated --enable-workload-identity install method and are still on v1.2.2. I can't seem to find any docs to say which AKS release version it will be included in, we are on 1.29.4 already so I will keep an eye out for updates.

**Comment by zadigus** [comment]
The problem is still there on AKS 1.30.2. I freshly deployed a new cluster with an application using workload identities together with istio and native sidecars. I am still getting the error

Pod "some-pod-name" is invalid:
        [spec.initContainers[1].lifecycle: Forbidden: may not be set for init
        containers without restartPolicy=Always,
        spec.initContainers[1].readinessProbe: Forbidden: may not be set for
        init containers without restartPolicy=Always,
        spec.initContainers[1].startupProbe: Forbidden: may not be set for init
        containers without restartPolicy=Always]
